doc/vlan/SONiC_VLAN_trunk_all_except_none_HLD.md

```diff
@@ -0,0 +1,701 @@
+             +--ro oper_status?          scommon:oper-status
+             +--ro admin_status?         scommon:admin-status
+             +--ro description?          string
+             +--ro speed?                uint64
+             +--ro port_load_interval?   uint16
+             +--ro fec?                  scommon:fec-mode
+
+
+module: sonic-portchannel
+    +--rw sonic-portchannel
+       +--rw PORTCHANNEL_GLOBAL
+       |  +--rw PORTCHANNEL_GLOBAL_LIST* [keyleaf]
+       |     +--rw keyleaf                    enumeration
+       |     +--rw graceful_shutdown_mode?    mode
+       +--rw PORTCHANNEL
+       |  +--rw PORTCHANNEL_LIST* [name]
+       |     +--rw name                       string
+       |     +--rw admin_status?              scommon:admin-status
+       |     +--rw mtu?                       uint32
+       |     +--rw static?                    boolean
+       |     +--rw min_links?                 uint8
+       |     +--rw fallback?                  boolean
+       |     +--rw fast_rate?                 boolean
+       |     +--rw graceful_shutdown_mode?    mode
+       |     +--rw description?               string
++      |     +--rw tagged_vlans*              string
++      |     +--rw access_vlan?               string
+       +--rw PORTCHANNEL_MEMBER
+       |  +--rw PORTCHANNEL_MEMBER_LIST* [name ifname]
+       |     +--rw name      -> ../../../PORTCHANNEL/PORTCHANNEL_LIST/name
+       |     +--rw ifname    -> /prt:sonic-port/PORT/PORT_LIST/ifname
+       +--ro LAG_TABLE
+       |  +--ro LAG_TABLE_LIST* [lagname]
+       |     +--ro lagname                 string
+       |     +--ro admin_status?           scommon:admin-status
+       |     +--ro mtu?                    uint32
+       |     +--ro active?                 boolean
+       |     +--ro name?                   string
+       |     +--ro oper_status?            scommon:oper-status
+       |     +--ro traffic_disable?        boolean
+       |     +--ro fallback_operational?   boolean
+       |     +--ro speed?                  uint64
+       |     +--ro port_load_interval?     uint16
+       +--ro LAG_MEMBER_TABLE
+          +--ro LAG_MEMBER_TABLE_LIST* [name ifname]
+             +--ro name      -> ../../../LAG_TABLE/LAG_TABLE_LIST/lagname
+             +--ro ifname    -> /prt:sonic-port/PORT/PORT_LIST/ifname
+             +--ro status?   string
+```
+```diff
+
+module: openconfig-interfaces-ext
+
+  ...
+
+  rpcs:
+    +---x clear-counters
+    |  +---w input
+    |  |  +---w interface-param?   string
+    |  +--ro output
+    |     +--ro status?          int32
+    |     +--ro status-detail?   string
++   +---x vlan-replace
++      +---w input
++      |  +---w ifname*     string
++      |  +---w vlanlist*   union
++      +--ro output
++         +--ro status?          uint32
++         +--ro status-detail?   string
+
+```
+
+### 3.3.2 CLI
+#### 3.3.2.1 Configuration Commands
+
+**switchport trunk allowed vlan {*vlan-list*|except *vlan-list* | none | all}**
```

Comment:
For completeness, we should show the add and remove options also (though these are existing ones).